blog/breach-to-barrier.md

```diff
@@ -0,0 +1,33 @@
+
+Now that we are all freshly aware that an interaction as innocent as “rendering an image” is a potentially dangerous activity, we want to use this opportunity to remind everyone that Electron comes with a sandbox that will limit the blast radius of the next big attack — whatever it may be.
+
+The memory sandbox was available ever since Electron v1 and enabled by default in v20, but we know that many apps (especially those that have been around for a while) may have a `sandbox: false` somewhere in their code. That’s understandable: If you’ve been with us for a long time, you probably enjoyed the power of throwing a `require("child_process")` or `require("fs")` into the same code that runs your HTML/CSS.
+
+Before we talk about _how_ you migrate to the sandbox, let’s first discuss _why_ you want it. The sandbox puts a hard cage around all renderer processes, ensuring that no matter what happens inside, code is executed inside a restricted environment. It gets to freely use CPU cycles and memory — that’s it. Sandboxed processes cannot write to disk or display their own windows. In the case of our `libwep` bug, the sandbox makes sure that an attacker cannot install or run malware. In fact, in the case of the original Pegasus attack on our employee’s iPhone, the attack specifically targeted a non-sandboxed image process to gain access to the phone. When a CVE like the one in this example is announced, you still have to upgrade your Electron apps to a secure version — but in the meantime, the amount of damage an attacker can do is limited dramatically.
+
+Migrating a vanilla Electron application with `sandbox: false` to `sandbox: true` is an undertaking. I know, because even though I have personally written the first draft of the [Electron Security Guidelines](https://www.electronjs.org/docs/latest/tutorial/security), I have not managed to migrate one of my own apps to use it. That changed this weekend, and I recommend that you change it, too.
+
+![Don’t be scared by the number of line changes, most of it is in `package-lock.json`](/assets/img/breach-to-barrier.png)
+
+There are two things you need to tackle:
+
+1. If you’re using Node.js code in either `preload` scripts or the actual `WebContents`, you need to move all that Node.js interaction to the main process (or, if you are fancy, a utility process). Given how powerful renderers have become, chances are high that the vast majority of your code doesn’t really need refactoring.
+
+Consult our documentation on [Inter-Process Communication](https://www.electronjs.org/docs/latest/tutorial/ipc). In my case, I moved a lot of code and wrapped it in `ipcRenderer.invoke()` and `ipcMain.handle()`, but the process was straightforward and quickly done. 2. Since enabling the sandbox disables Node.js integration in your preload scripts, you can no longer use `require("../my-script")`. In other words, your preload script needs to be a single file.
```

Comment:
Nit: put the second point on its own line.
```suggestion
Consult our documentation on [Inter-Process Communication](https://www.electronjs.org/docs/latest/tutorial/ipc). In my case, I moved a lot of code and wrapped it in `ipcRenderer.invoke()` and `ipcMain.handle()`, but the process was straightforward and quickly done.

2. Since enabling the sandbox disables Node.js integration in your preload scripts, you can no longer use `require("../my-script")`. In other words, your preload script needs to be a single file.
```